mapper/thin_edge_json/src/lib.rs

```diff
@@ -0,0 +1,574 @@
+        timestamp: DateTime<Utc>,
+    ) -> Result<ThinEdgeJson, ThinEdgeJsonError> {
+        let mut measurements = vec![];
+        let mut time = timestamp.to_rfc3339();
+        match &input {
+            JsonValue::Object(thin_edge_obj) => {
+                for (k, v) in thin_edge_obj.iter() {
+                    if k.eq("type") {
+                        return Err(ThinEdgeJsonError::ThinEdgeReservedWordError {
+                            name: String::from(k),
+                        });
+                    } else if k.eq("time") {
+                        time = ThinEdgeJson::check_timestamp_for_iso8601_complaint(v)?;
+                    } else {
+                        match v {
+                            //Single Value object
+                            JsonValue::Number(num) => {
+                                let single_value_measurement =
+                                    SingleValueMeasurement::new(k, (*num).into())?;
+                                measurements.push(ThinEdgeValue::Single(single_value_measurement));
+                            }
+                            //Multi value object
+                            JsonValue::Object(multi_value_thin_edge_object) => {
+                                let multi_value_measurement =
+                                    MultiValueMeasurement::new(k, multi_value_thin_edge_object)?;
+                                measurements.push(ThinEdgeValue::Multi(multi_value_measurement));
+                            }
+
+                            _ => {
+                                return Err(ThinEdgeJsonError::new_invalid_json_value(k, v));
+                            }
+                        }
+                    }
+                }
+                if measurements.is_empty() {
+                    Err(ThinEdgeJsonError::EmptyThinEdgeJsonRoot)
+                } else {
+                    Ok(ThinEdgeJson {
+                        timestamp: time,
+                        values: measurements,
+                    })
+                }
+            }
+            _ => Err(ThinEdgeJsonError::new_invalid_json_root(&input)),
+        }
+    }
+
+    fn check_timestamp_for_iso8601_complaint(
+        value: &JsonValue,
+    ) -> Result<String, ThinEdgeJsonError> {
+        match value {
+            JsonValue::Short(str) => {
+                let timestamp = str.as_str();
+                //Parse fails if timestamp is not is8601 complaint
+                DateTime::parse_from_rfc3339(&timestamp).map_err(|err| {
+                    ThinEdgeJsonError::InvalidTimestamp {
+                        value: String::from(timestamp),
+                        from: err,
+                    }
+                })?;
+                Ok(String::from(timestamp))
+            }
+            _ => Err(ThinEdgeJsonError::new_invalid_json_time(value)),
+        }
+    }
+
+    pub fn serialize_thin_edge(&self) -> Vec<u8> {
+        serde_json::to_vec(self).unwrap()
+    }
```

Comment:
What is the layout of the JSON output? Unlikely some ThinEdgeJson.

* This function has to be removed or fixed.